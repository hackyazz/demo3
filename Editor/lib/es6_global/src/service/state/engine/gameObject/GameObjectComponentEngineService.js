

import * as GameObjectAPI$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/api/GameObjectAPI.js";
import * as GetComponentGameObjectService$Wonderjs from "../../../../../../../node_modules/wonder.js/lib/es6_global/src/service/record/main/gameObject/GetComponentGameObjectService.js";

function getLightMaterialComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getLightMaterialComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

function getBasicMaterialComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getBasicMaterialComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

function getMeshRendererComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getMeshRendererComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

var unsafeGetGeometryComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectGeometryComponent;

function getGeometryComponent(gameObject, engineState) {
  return GetComponentGameObjectService$Wonderjs.getGeometryComponent(gameObject, engineState[/* gameObjectRecord */10]);
}

function hasLightComponent(gameObject, engineState) {
  if (GameObjectAPI$Wonderjs.hasGameObjectDirectionLightComponent(gameObject, engineState)) {
    return true;
  } else {
    return GameObjectAPI$Wonderjs.hasGameObjectPointLightComponent(gameObject, engineState);
  }
}

var hasLightMaterialComponent = GameObjectAPI$Wonderjs.hasGameObjectLightMaterialComponent;

var unsafeGetLightMaterialComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectLightMaterialComponent;

var addLightMaterialComponent = GameObjectAPI$Wonderjs.addGameObjectLightMaterialComponent;

var disposeLightMaterialComponent = GameObjectAPI$Wonderjs.disposeGameObjectLightMaterialComponent;

var removeLightMaterialComponent = GameObjectAPI$Wonderjs.removeGameObjectLightMaterialComponent;

var hasBasicMaterialComponent = GameObjectAPI$Wonderjs.hasGameObjectBasicMaterialComponent;

var unsafeGetBasicMaterialComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectBasicMaterialComponent;

var addBasicMaterialComponent = GameObjectAPI$Wonderjs.addGameObjectBasicMaterialComponent;

var disposeBasicMaterialComponent = GameObjectAPI$Wonderjs.disposeGameObjectBasicMaterialComponent;

var removeBasicMaterialComponent = GameObjectAPI$Wonderjs.removeGameObjectBasicMaterialComponent;

var addMeshRendererComponent = GameObjectAPI$Wonderjs.addGameObjectMeshRendererComponent;

var unsafeGetMeshRendererComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectMeshRendererComponent;

var hasMeshRendererComponent = GameObjectAPI$Wonderjs.hasGameObjectMeshRendererComponent;

var disposeMeshRendererComponent = GameObjectAPI$Wonderjs.disposeGameObjectMeshRendererComponent;

var disposeGeometryComponent = GameObjectAPI$Wonderjs.disposeGameObjectGeometryComponent;

var removeGeometryComponent = GameObjectAPI$Wonderjs.removeGameObjectGeometryComponent;

var hasGeometryComponent = GameObjectAPI$Wonderjs.hasGameObjectGeometryComponent;

var addGeometryComponent = GameObjectAPI$Wonderjs.addGameObjectGeometryComponent;

var addPerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.addGameObjectPerspectiveCameraProjectionComponent;

var hasPerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.hasGameObjectPerspectiveCameraProjectionComponent;

var unsafeGetPerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectPerspectiveCameraProjectionComponent;

var disposePerspectiveCameraProjectionComponent = GameObjectAPI$Wonderjs.disposeGameObjectPerspectiveCameraProjectionComponent;

var addBasicCameraViewComponent = GameObjectAPI$Wonderjs.addGameObjectBasicCameraViewComponent;

var hasBasicCameraViewComponent = GameObjectAPI$Wonderjs.hasGameObjectBasicCameraViewComponent;

var unsafeGetBasicCameraViewComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectBasicCameraViewComponent;

var disposeBasicCameraViewComponent = GameObjectAPI$Wonderjs.disposeGameObjectBasicCameraViewComponent;

var unsafeGetTransformComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectTransformComponent;

var hasTransformComponent = GameObjectAPI$Wonderjs.hasGameObjectTransformComponent;

var unsafeGetSourceInstanceComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectSourceInstanceComponent;

var hasSourceInstanceComponent = GameObjectAPI$Wonderjs.hasGameObjectSourceInstanceComponent;

var addSourceInstanceComponent = GameObjectAPI$Wonderjs.addGameObjectSourceInstanceComponent;

var addDirectionLightComponent = GameObjectAPI$Wonderjs.addGameObjectDirectionLightComponent;

var disposeDirectionLightComponent = GameObjectAPI$Wonderjs.disposeGameObjectDirectionLightComponent;

var unsafeGetDirectionLightComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectDirectionLightComponent;

var hasDirectionLightComponent = GameObjectAPI$Wonderjs.hasGameObjectDirectionLightComponent;

var addPointLightComponent = GameObjectAPI$Wonderjs.addGameObjectPointLightComponent;

var disposePointLightComponent = GameObjectAPI$Wonderjs.disposeGameObjectPointLightComponent;

var unsafeGetPointLightComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectPointLightComponent;

var hasPointLightComponent = GameObjectAPI$Wonderjs.hasGameObjectPointLightComponent;

var addArcballCameraControllerComponent = GameObjectAPI$Wonderjs.addGameObjectArcballCameraControllerComponent;

var disposeArcballCameraControllerComponent = GameObjectAPI$Wonderjs.disposeGameObjectArcballCameraControllerComponent;

var unsafeGetArcballCameraControllerComponent = GameObjectAPI$Wonderjs.unsafeGetGameObjectArcballCameraControllerComponent;

var hasArcballCameraControllerComponent = GameObjectAPI$Wonderjs.hasGameObjectArcballCameraControllerComponent;

var getAllLightMaterialComponents = GameObjectAPI$Wonderjs.getAllLightMaterialComponents;

var getAllDirectionLightComponents = GameObjectAPI$Wonderjs.getAllDirectionLightComponents;

var getAllBasicCameraViewComponents = GameObjectAPI$Wonderjs.getAllBasicCameraViewComponents;

var getAllArcballCameraControllerComponents = GameObjectAPI$Wonderjs.getAllArcballCameraControllerComponents;

var getAllPerspectiveCameraProjectionComponents = GameObjectAPI$Wonderjs.getAllPerspectiveCameraProjectionComponents;

var getAllGeometryComponents = GameObjectAPI$Wonderjs.getAllGeometryComponents;

export {
  hasLightMaterialComponent ,
  getLightMaterialComponent ,
  unsafeGetLightMaterialComponent ,
  addLightMaterialComponent ,
  disposeLightMaterialComponent ,
  removeLightMaterialComponent ,
  hasBasicMaterialComponent ,
  getBasicMaterialComponent ,
  unsafeGetBasicMaterialComponent ,
  addBasicMaterialComponent ,
  disposeBasicMaterialComponent ,
  removeBasicMaterialComponent ,
  addMeshRendererComponent ,
  getMeshRendererComponent ,
  unsafeGetMeshRendererComponent ,
  hasMeshRendererComponent ,
  disposeMeshRendererComponent ,
  unsafeGetGeometryComponent ,
  getGeometryComponent ,
  disposeGeometryComponent ,
  removeGeometryComponent ,
  hasGeometryComponent ,
  addGeometryComponent ,
  addPerspectiveCameraProjectionComponent ,
  hasPerspectiveCameraProjectionComponent ,
  unsafeGetPerspectiveCameraProjectionComponent ,
  disposePerspectiveCameraProjectionComponent ,
  addBasicCameraViewComponent ,
  hasBasicCameraViewComponent ,
  unsafeGetBasicCameraViewComponent ,
  disposeBasicCameraViewComponent ,
  unsafeGetTransformComponent ,
  hasTransformComponent ,
  unsafeGetSourceInstanceComponent ,
  hasSourceInstanceComponent ,
  addSourceInstanceComponent ,
  addDirectionLightComponent ,
  disposeDirectionLightComponent ,
  unsafeGetDirectionLightComponent ,
  hasDirectionLightComponent ,
  addPointLightComponent ,
  disposePointLightComponent ,
  unsafeGetPointLightComponent ,
  hasPointLightComponent ,
  hasLightComponent ,
  addArcballCameraControllerComponent ,
  disposeArcballCameraControllerComponent ,
  unsafeGetArcballCameraControllerComponent ,
  hasArcballCameraControllerComponent ,
  getAllLightMaterialComponents ,
  getAllDirectionLightComponents ,
  getAllBasicCameraViewComponents ,
  getAllArcballCameraControllerComponents ,
  getAllPerspectiveCameraProjectionComponents ,
  getAllGeometryComponents ,
  
}
/* GameObjectAPI-Wonderjs Not a pure module */
