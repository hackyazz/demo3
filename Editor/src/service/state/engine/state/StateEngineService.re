let deepCopyForRestore = Wonderjs.StateAPI.deepCopyForRestore;

let createState = Wonderjs.StateAPI.createState;

let createStateData = Wonderjs.StateAPI.createStateData;

let getStateFromData = Wonderjs.StateAPI.getStateFromData;

let unsafeGetState = Wonderjs.StateAPI.unsafeGetState;

let setStateToData = Wonderjs.StateAPI.setStateToData;

let setState = Wonderjs.StateAPI.setState;

let setIsDebug = Wonderjs.StateAPI.setIsDebug;