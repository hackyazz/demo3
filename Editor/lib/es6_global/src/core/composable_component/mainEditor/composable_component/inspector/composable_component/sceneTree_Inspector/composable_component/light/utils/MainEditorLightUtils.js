

import * as Curry from "../../../../../../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Log$WonderLog from "../../../../../../../../../../../../../node_modules/wonder-log/lib/es6_global/src/Log.js";
import * as LogUtils$WonderEditor from "../../../../../../../../../utils/console/LogUtils.js";
import * as ConsoleUtils$WonderEditor from "../../../../../../../../../utils/ui/ConsoleUtils.js";
import * as ArrayService$WonderCommonlib from "../../../../../../../../../../../../../node_modules/wonder-commonlib/lib/es6_global/src/ArrayService.js";
import * as StateLogicService$WonderEditor from "../../../../../../../../../../service/stateTuple/logic/StateLogicService.js";
import * as SceneEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/SceneEngineService.js";
import * as StateEditorService$WonderEditor from "../../../../../../../../../../service/state/editor/StateEditorService.js";
import * as StateEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/state/StateEngineService.js";
import * as GameObjectLogicService$WonderEditor from "../../../../../../../../../../service/stateTuple/logic/GameObjectLogicService.js";
import * as SceneTreeEditorService$WonderEditor from "../../../../../../../../../../service/state/editor/sceneTree/SceneTreeEditorService.js";
import * as GameObjectEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/gameObject/GameObjectEngineService.js";
import * as PointLightEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/PointLightEngineService.js";
import * as DirectionLightEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/DirectionLightEngineService.js";
import * as OperatePointLightLogicService$WonderEditor from "../../../../../../../../../../service/stateTuple/logic/OperatePointLightLogicService.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/gameObject/GameObjectComponentEngineService.js";
import * as HierarchyGameObjectEngineService$WonderEditor from "../../../../../../../../../../service/state/engine/gameObject/HierarchyGameObjectEngineService.js";
import * as OperateDirectionLightLogicService$WonderEditor from "../../../../../../../../../../service/stateTuple/logic/OperateDirectionLightLogicService.js";

function getLightOptions(param) {
  return /* array */[
          /* record */[
            /* key : DirectionLight */0,
            /* value */"direction_light"
          ],
          /* record */[
            /* key : PointLight */1,
            /* value */"point_light"
          ]
        ];
}

function isLightGameObject(gameObject, engineState) {
  if (GameObjectComponentEngineService$WonderEditor.hasDirectionLightComponent(gameObject, engineState)) {
    return true;
  } else {
    return GameObjectComponentEngineService$WonderEditor.hasPointLightComponent(gameObject, engineState);
  }
}

function getLightTypeByGameObject(gameObject, engineState) {
  var match = GameObjectComponentEngineService$WonderEditor.hasDirectionLightComponent(gameObject, engineState);
  var match$1 = GameObjectComponentEngineService$WonderEditor.hasPointLightComponent(gameObject, engineState);
  var exit = 0;
  if (match) {
    if (match$1) {
      exit = 1;
    } else {
      return /* DirectionLight */0;
    }
  } else if (match$1) {
    return /* PointLight */1;
  } else {
    exit = 1;
  }
  if (exit === 1) {
    return Log$WonderLog.fatal(LogUtils$WonderEditor.buildFatalMessage("gameObject:" + (String(gameObject) + " should has light component"), "", "", ""));
  }
  
}

function handleSpecificFuncByLightType(lightType, param) {
  var currentSceneTreeNode = StateLogicService$WonderEditor.getEditorState(SceneTreeEditorService$WonderEditor.unsafeGetCurrentSceneTreeNode);
  if (lightType) {
    return Curry._1(param[1], currentSceneTreeNode);
  } else {
    return Curry._1(param[0], currentSceneTreeNode);
  }
}

function getDirectionLightExceedMaxCountMessage(param) {
  return "the direction light count is exceed max count!";
}

function getPointLightExceedMaxCountMessage(param) {
  return "the point light count is exceed max count!";
}

function isLightExceedMaxCountByType(targetLightType, engineState) {
  if (targetLightType) {
    return /* tuple */[
            "the point light count is exceed max count!",
            PointLightEngineService$WonderEditor.isMaxCount(engineState)
          ];
  } else {
    return /* tuple */[
            "the direction light count is exceed max count!",
            DirectionLightEngineService$WonderEditor.isMaxCount(engineState)
          ];
  }
}

function isLightExceedMaxCountByCount(param, engineState) {
  var match = DirectionLightEngineService$WonderEditor.isMaxCountByCount(param[0], engineState);
  if (match) {
    return /* tuple */[
            "the direction light count is exceed max count!",
            true
          ];
  } else {
    var match$1 = PointLightEngineService$WonderEditor.isMaxCountByCount(param[1], engineState);
    if (match$1) {
      return /* tuple */[
              "the point light count is exceed max count!",
              true
            ];
    } else {
      return /* tuple */[
              "",
              false
            ];
    }
  }
}

function getLightCount(gameObject, engineState) {
  return ArrayService$WonderCommonlib.reduceOneParam((function (param, gameObject) {
                var pointLightCount = param[1];
                var directionLightCount = param[0];
                var match = GameObjectComponentEngineService$WonderEditor.hasDirectionLightComponent(gameObject, engineState);
                var match$1 = GameObjectComponentEngineService$WonderEditor.hasPointLightComponent(gameObject, engineState);
                return /* tuple */[
                        match ? directionLightCount + 1 | 0 : directionLightCount,
                        match$1 ? pointLightCount + 1 | 0 : pointLightCount
                      ];
              }), /* tuple */[
              0,
              0
            ], HierarchyGameObjectEngineService$WonderEditor.getAllGameObjects(gameObject, engineState));
}

function _getOperateSourceLightFunc(lightType, gameObject, engineState) {
  if (lightType) {
    return /* tuple */[
            GameObjectComponentEngineService$WonderEditor.unsafeGetPointLightComponent(gameObject, engineState),
            OperatePointLightLogicService$WonderEditor.disposePointLight
          ];
  } else {
    return /* tuple */[
            GameObjectComponentEngineService$WonderEditor.unsafeGetDirectionLightComponent(gameObject, engineState),
            OperateDirectionLightLogicService$WonderEditor.disposeDirectionLight
          ];
  }
}

function _getOperateTargetLightFunc(lightType, engineState) {
  if (lightType) {
    return /* tuple */[
            OperatePointLightLogicService$WonderEditor.createPointLight(engineState),
            OperatePointLightLogicService$WonderEditor.addPointLight
          ];
  } else {
    return /* tuple */[
            OperateDirectionLightLogicService$WonderEditor.createDirectionLight(engineState),
            OperateDirectionLightLogicService$WonderEditor.addDirectionLight
          ];
  }
}

function replaceLightByType(sourceLightType, targetLightType) {
  var gameObject = StateLogicService$WonderEditor.getEditorState(SceneTreeEditorService$WonderEditor.unsafeGetCurrentSceneTreeNode);
  var editorState = StateEditorService$WonderEditor.getState(/* () */0);
  var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
  var match = isLightExceedMaxCountByType(targetLightType, engineState);
  if (match[1]) {
    return ConsoleUtils$WonderEditor.warn(match[0], editorState);
  } else {
    var match$1 = _getOperateSourceLightFunc(sourceLightType, gameObject, engineState);
    var match$2 = _getOperateTargetLightFunc(targetLightType, engineState);
    var match$3 = match$2[0];
    var engineState$1 = GameObjectEngineService$WonderEditor.initGameObject(gameObject, Curry._3(match$2[1], gameObject, match$3[1], Curry._3(match$1[1], gameObject, match$1[0], match$3[0])));
    StateLogicService$WonderEditor.refreshEngineState(engineState$1);
    return StateLogicService$WonderEditor.getAndRefreshEngineStateWithFunc(SceneEngineService$WonderEditor.clearShaderCacheAndReInitAllLightMaterials);
  }
}

function disposeLightByLightType(lightType, currentSceneTreeNode, param) {
  var engineState = param[1];
  var editorState = param[0];
  if (lightType) {
    return GameObjectLogicService$WonderEditor.disposePointLight(currentSceneTreeNode, GameObjectComponentEngineService$WonderEditor.unsafeGetPointLightComponent(currentSceneTreeNode, engineState), /* tuple */[
                editorState,
                engineState
              ]);
  } else {
    return GameObjectLogicService$WonderEditor.disposeDirectionLight(currentSceneTreeNode, GameObjectComponentEngineService$WonderEditor.unsafeGetDirectionLightComponent(currentSceneTreeNode, engineState), /* tuple */[
                editorState,
                engineState
              ]);
  }
}

export {
  getLightOptions ,
  isLightGameObject ,
  getLightTypeByGameObject ,
  handleSpecificFuncByLightType ,
  getDirectionLightExceedMaxCountMessage ,
  getPointLightExceedMaxCountMessage ,
  isLightExceedMaxCountByType ,
  isLightExceedMaxCountByCount ,
  getLightCount ,
  _getOperateSourceLightFunc ,
  _getOperateTargetLightFunc ,
  replaceLightByType ,
  disposeLightByLightType ,
  
}
/* Log-WonderLog Not a pure module */
