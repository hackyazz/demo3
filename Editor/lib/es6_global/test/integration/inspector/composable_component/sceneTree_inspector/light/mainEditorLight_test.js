

import * as Curry from "../../../../../../../../node_modules/bs-platform/lib/es6/curry.js";
import * as Sinon from "../../../../../../../../node_modules/wonder-bs-sinon/lib/es6_global/src/sinon.js";
import * as Sinon$1 from "sinon";
import * as Wonder_jest from "../../../../../../../../node_modules/wonder-bs-jest/lib/es6_global/src/wonder_jest.js";
import * as ConsoleTool$WonderEditor from "../../../../../unit/tool/external/ConsoleTool.js";
import * as ReactTestTool$WonderEditor from "../../../../../tool/ReactTestTool.js";
import * as GLSLToolEngine$WonderEditor from "../../../../../tool/engine/GLSLToolEngine.js";
import * as GameObjectTool$WonderEditor from "../../../../../tool/GameObjectTool.js";
import * as FakeGlToolEngine$WonderEditor from "../../../../../tool/engine/FakeGlToolEngine.js";
import * as EventListenerTool$WonderEditor from "../../../../../unit/tool/EventListenerTool.js";
import * as StateLogicService$WonderEditor from "../../../../../../src/service/stateTuple/logic/StateLogicService.js";
import * as BuildComponentTool$WonderEditor from "../../../../../tool/BuildComponentTool.js";
import * as StateEngineService$WonderEditor from "../../../../../../src/service/state/engine/StateEngineService.js";
import * as MainEditorLightTool$WonderEditor from "./tool/MainEditorLightTool.js";
import * as MainEditorSceneTool$WonderEditor from "../../../../../tool/MainEditorSceneTool.js";
import * as MainEditorLightUtils$WonderEditor from "../../../../../../src/core/composable_component/mainEditor/composable_component/inspector/composable_component/sceneTree_Inspector/composable_component/light/utils/MainEditorLightUtils.js";
import * as PointLightToolEngine$WonderEditor from "../../../../../tool/engine/PointLightToolEngine.js";
import * as NoWorkerJobConfigToolEngine$WonderEditor from "../../../../../tool/engine/NoWorkerJobConfigToolEngine.js";
import * as GameObjectComponentEngineService$WonderEditor from "../../../../../../src/service/state/engine/GameObjectComponentEngineService.js";

describe("MainEditorLight component", (function () {
        var sandbox = Sinon.getSandboxDefaultVal(/* () */0);
        beforeEach((function () {
                sandbox[0] = Sinon$1.sandbox.create();
                MainEditorSceneTool$WonderEditor.initStateWithJob(sandbox, NoWorkerJobConfigToolEngine$WonderEditor.buildNoWorkerJobConfig(undefined, undefined, "\n                   [\n                       {\n                           \"name\": \"default\",\n                           \"jobs\": [\n                               {\n                                   \"name\": \"dispose\"\n                               }\n                           ]\n                       }\n                   ]\n               ", undefined, undefined, /* () */0), undefined, undefined, undefined, /* () */0);
                return Curry._1(EventListenerTool$WonderEditor.stubGetElementByIdReturnFakeDom, Curry._1(EventListenerTool$WonderEditor.buildFakeDom, /* () */0));
              }));
        afterEach((function () {
                return Curry._1(Sinon.restoreSandbox, sandbox[0]);
              }));
        describe("test set currentSceneTreeNode", (function () {
                beforeEach((function () {
                        return MainEditorSceneTool$WonderEditor.createDefaultScene(sandbox, MainEditorSceneTool$WonderEditor.setDirectionLightGameObjectToBeCurrentSceneTreeNode);
                      }));
                describe("test change light type", (function () {
                        Wonder_jest.test("test the original type is direction light, should show direction light component ", (function () {
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildLight(/* () */0));
                              }));
                        Wonder_jest.test("test change type to be point light, should show point light component", (function () {
                                MainEditorLightTool$WonderEditor.setLightTypeToBePointLight(/* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildLight(/* () */0));
                              }));
                        Wonder_jest.test("test change type to be direction light, should show direction light component", (function () {
                                MainEditorLightTool$WonderEditor.setLightTypeToBePointLight(/* () */0);
                                MainEditorLightTool$WonderEditor.setLightTypeToBeDirectionLight(/* () */0);
                                return ReactTestTool$WonderEditor.createSnapshotAndMatch(BuildComponentTool$WonderEditor.buildLight(/* () */0));
                              }));
                        describe("should re-init all light material components in the scene", (function () {
                                describe("test change direction light to be point light", (function () {
                                        return Wonder_jest.test("test\n               glsl->DIRECTION_LIGHTS_COUNT should - 1;\n               glsl->POINT_LIGHTS_COUNT should + 1;", (function () {
                                                      var gl = FakeGlToolEngine$WonderEditor.getEngineStateGl(/* () */0);
                                                      var glShaderSource = gl.shaderSource;
                                                      MainEditorLightTool$WonderEditor.setLightTypeToBePointLight(/* () */0);
                                                      return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](/* tuple */[
                                                                      GLSLToolEngine$WonderEditor.contain(GLSLToolEngine$WonderEditor.getVsSource(glShaderSource), "#define DIRECTION_LIGHTS_COUNT 0"),
                                                                      GLSLToolEngine$WonderEditor.contain(GLSLToolEngine$WonderEditor.getVsSource(glShaderSource), "#define POINT_LIGHTS_COUNT 1")
                                                                    ]), /* tuple */[
                                                                  true,
                                                                  true
                                                                ]);
                                                    }));
                                      }));
                                return /* () */0;
                              }));
                        describe("deal with specific cases", (function () {
                                beforeEach((function () {
                                        return ConsoleTool$WonderEditor.notShowMessage(/* () */0);
                                      }));
                                afterEach((function () {
                                        return ConsoleTool$WonderEditor.showMessage(/* () */0);
                                      }));
                                return Wonder_jest.test("if target light type is maxCount, not change to it and warn", (function () {
                                              var engineState = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                              var match = PointLightToolEngine$WonderEditor.createGameObject(engineState);
                                              var match$1 = PointLightToolEngine$WonderEditor.createGameObject(match[0]);
                                              var match$2 = PointLightToolEngine$WonderEditor.createGameObject(match$1[0]);
                                              PointLightToolEngine$WonderEditor.createGameObject(match$2[0]);
                                              var warn = Sinon.createMethodStubWithJsObjSandbox(sandbox, console, "warn");
                                              MainEditorLightTool$WonderEditor.setLightTypeToBePointLight(/* () */0);
                                              var engineState$1 = StateEngineService$WonderEditor.unsafeGetState(/* () */0);
                                              var __x = GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0);
                                              return Curry._2(Wonder_jest.Expect[/* Operators */23][/* = */5], Wonder_jest.Expect[/* expect */0](/* tuple */[
                                                              ConsoleTool$WonderEditor.getMessage(warn).includes("the point light count is exceed max count"),
                                                              GameObjectComponentEngineService$WonderEditor.hasPointLightComponent(__x, engineState$1)
                                                            ]), /* tuple */[
                                                          true,
                                                          false
                                                        ]);
                                            }));
                              }));
                        return /* () */0;
                      }));
                describe("deal with specific cases", (function () {
                        return Wonder_jest.test("test getLightTypeByGameObject should throw error when gameObject haven't light ", (function () {
                                      return Wonder_jest.Expect[/* toThrowMessageRe */21]((/should\shas\slight\scomponent/img), Wonder_jest.Expect[/* expect */0]((function () {
                                                        MainEditorSceneTool$WonderEditor.setFirstCubeToBeCurrentSceneTreeNode(/* () */0);
                                                        var partial_arg = GameObjectTool$WonderEditor.unsafeGetCurrentSceneTreeNode(/* () */0);
                                                        return StateLogicService$WonderEditor.getEngineStateToGetData((function (param) {
                                                                      return MainEditorLightUtils$WonderEditor.getLightTypeByGameObject(partial_arg, param);
                                                                    }));
                                                      })));
                                    }));
                      }));
                return /* () */0;
              }));
        return /* () */0;
      }));

export {
  
}
/*  Not a pure module */
