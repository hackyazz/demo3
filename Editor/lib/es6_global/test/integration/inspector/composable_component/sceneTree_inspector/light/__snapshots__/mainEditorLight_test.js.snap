// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorLight component test set currentSceneTreeNode test change light type test change type to be direction light, should show direction light component 1`] = `
<article
  className="wonder-inspector-light"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Type
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          direction_light
        </option>
        <option
          value="1"
        >
          point_light
        </option>
      </select>
    </div>
  </article>
  <article
    className="wonder-direction-light"
  >
    <article
      className="inspector-item"
    >
      <div
        className="item-header"
      >
        Color
      </div>
      <div
        className="item-content item-color"
      >
        <div
          className="color-hex"
          onClick={[Function]}
          style={
            Object {
              "background": "#ffffff",
            }
          }
        />
        <div
          className="color-select"
          onClick={[Function]}
        >
          <img
            src="./public/img/color.png"
          />
        </div>
      </div>
    </article>
    <article
      className="inspector-item wonder-float-input"
    >
      <div
        className="item-header component-label"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
      >
        Intensity
      </div>
      <div
        className="item-content"
      >
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </div>
    </article>
  </article>
</article>
`;

exports[`MainEditorLight component test set currentSceneTreeNode test change light type test change type to be point light, should show point light component 1`] = `
<article
  className="wonder-inspector-light"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Type
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="1"
      >
        <option
          value="0"
        >
          direction_light
        </option>
        <option
          value="1"
        >
          point_light
        </option>
      </select>
    </div>
  </article>
  <article
    className="wonder-point-light"
  >
    <article
      className="inspector-item"
    >
      <div
        className="item-header"
      >
        Color
      </div>
      <div
        className="item-content item-color"
      >
        <div
          className="color-hex"
          onClick={[Function]}
          style={
            Object {
              "background": "#ffffff",
            }
          }
        />
        <div
          className="color-select"
          onClick={[Function]}
        >
          <img
            src="./public/img/color.png"
          />
        </div>
      </div>
    </article>
    <article
      className="inspector-item wonder-float-input"
    >
      <div
        className="item-header component-label"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
      >
        Intensity
      </div>
      <div
        className="item-content"
      >
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </div>
    </article>
    <article
      className="inspector-item wonder-float-input"
    >
      <div
        className="item-header component-label"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
      >
        Constant
      </div>
      <div
        className="item-content"
      >
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </div>
    </article>
    <article
      className="inspector-item wonder-float-input"
    >
      <div
        className="item-header component-label"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
      >
        Linear
      </div>
      <div
        className="item-content"
      >
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.07"
        />
      </div>
    </article>
    <article
      className="inspector-item wonder-float-input"
    >
      <div
        className="item-header component-label"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
      >
        Quadratic
      </div>
      <div
        className="item-content"
      >
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="0.017"
        />
      </div>
    </article>
    <article
      className="inspector-item wonder-float-input"
    >
      <div
        className="item-header component-label"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
      >
        Range
      </div>
      <div
        className="item-content"
      >
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="65"
        />
      </div>
    </article>
  </article>
</article>
`;

exports[`MainEditorLight component test set currentSceneTreeNode test change light type test the original type is direction light, should show direction light component  1`] = `
<article
  className="wonder-inspector-light"
>
  <article
    className="inspector-item"
  >
    <div
      className="item-header"
    >
      <span
        className=""
      >
        Type
      </span>
    </div>
    <div
      className="item-content"
    >
      <select
        onChange={[Function]}
        value="0"
      >
        <option
          value="0"
        >
          direction_light
        </option>
        <option
          value="1"
        >
          point_light
        </option>
      </select>
    </div>
  </article>
  <article
    className="wonder-direction-light"
  >
    <article
      className="inspector-item"
    >
      <div
        className="item-header"
      >
        Color
      </div>
      <div
        className="item-content item-color"
      >
        <div
          className="color-hex"
          onClick={[Function]}
          style={
            Object {
              "background": "#ffffff",
            }
          }
        />
        <div
          className="color-select"
          onClick={[Function]}
        >
          <img
            src="./public/img/color.png"
          />
        </div>
      </div>
    </article>
    <article
      className="inspector-item wonder-float-input"
    >
      <div
        className="item-header component-label"
        onMouseDown={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
      >
        Intensity
      </div>
      <div
        className="item-content"
      >
        <input
          className="input-component float-input"
          onBlur={[Function]}
          onChange={[Function]}
          type="text"
          value="1"
        />
      </div>
    </article>
  </article>
</article>
`;
